internal/services/storagecache/managed_lustre_file_system_auto_export_job_resource.go

```diff
@@ -0,0 +1,233 @@
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+
+		"admin_status": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(autoexportjob.AutoExportJobAdminStatusEnable),
+				string(autoexportjob.AutoExportJobAdminStatusDisable),
+			}, false),
+			Default: string(autoexportjob.AutoExportJobAdminStatusEnable),
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ManagedLustreFileSystemAutoExportJobModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding model: %w", err)
+			}
+
+			autoExportJobsClient := metadata.Client.StorageCache.AutoExportJobs
+			autoExportJobClient := metadata.Client.StorageCache.AutoExportJob
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			id := autoexportjobs.NewAutoExportJobID(subscriptionId, model.ResourceGroupName, model.AmlFileSystemName, model.Name)
+
+			existing, err := autoExportJobsClient.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing Auto Export Job %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			props := autoexportjob.AutoExportJob{
+				Location: location.Normalize(model.Location),
+				Properties: pointer.To(autoexportjob.AutoExportJobProperties{
+					AdminStatus:        pointer.To(autoexportjob.AutoExportJobAdminStatus(model.AdminStatus)),
+					AutoExportPrefixes: pointer.To(model.AutoExportPrefixes),
+				}),
+				Tags: pointer.To(model.Tags),
+			}
+
+			autoExportJobId := autoexportjob.NewAutoExportJobID(subscriptionId, model.ResourceGroupName, model.AmlFileSystemName, model.Name)
+			if err := autoExportJobClient.CreateOrUpdateThenPoll(ctx, autoExportJobId, props); err != nil {
+				return fmt.Errorf("creating Auto Export Job %s: %+v", id, err)
+			}
+
+			metadata.SetID(autoExportJobId)
+			return nil
+		},
+	}
+}
+
+func (r ManagedLustreFileSystemAutoExportJobResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.StorageCache.AutoExportJobs
+
+			id, err := autoexportjobs.ParseAutoExportJobID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving Auto Export Job %s: %+v", id, err)
```

Comment:
Same here with the error wrapping...

```suggestion
				return fmt.Errorf("retrieving Auto Export Job %s: %w", id, err)
```